site/_posts/2022-10-30-spring-and-fusionAuth.md

```diff
@@ -0,0 +1,353 @@
+
+Click the Save button at the top right for your changes to take effect.
+
+## Setting up OpenID Connect (OIDC)
+
+Once the user has logged in via the FusionAuth application, we can retrieve their FusionAuth profile using the [OIDC](https://fusionauth.io/docs/v1/tech/oauth/) functionality provided by FusionAuth.
+
+To configure this to work properly with Spring, we need to set a couple JWT parameters in FusionAuth.
+
+First, head to your Tenant configuration in FusionAuth. If you haven't made and linked a custom Tenant for your application, you can edit the default Tenant. Under the "General" tab in your Tenant settings, set the "Issuer" field to a URL representing your domain. Note that it must be a fully qualified URL for Spring to process it correctly (i.e. include the protocol). This issuer field is used in the [JWT](https://jwt.io) created by FusionAuth when logging in.
+
+{% include _image.liquid src="/assets/img/blogs/spring-fusionauth/fusionauth-issuer-url.png" alt="Configuring the JWT issuer URL in FusionAuth." class="img-fluid" figure=false %}
+
+Then head over to your FusionAuth application and edit it. Choose the "JWT" tab on the settings page, and enable JWT by setting the toggle to on. Then configure the JSON web tokens with the following values:
+
+- "JWT duration": 3600
+- "Access Token signing key": Access token signing key generated for application <name> (RS256)
+- "Id Token signing key": Access token signing key generated for application <name> (RS256)
+
+Click the "Save" button in the top right to save these changes. Your setup should look similar to this:
+
+{% include _image.liquid src="/assets/img/blogs/spring-fusionauth/fusionauth-jwt-config.png" alt="Configuring the JWT algorithm settings." class="img-fluid" figure=false %}
+
+## Setting up PKCE
+
+[Proof Key for Code Exchange (PKCE)](https://oauth.net/2/pkce/) was originally intended for public clients like native mobile or desktop apps that couldn't safely store a client secret. Now it is [recommended](https://oauth.net/2/grant-types/authorization-code/) even on confidential clients like web apps where the Client Secret is under secure control to protect against authorization code injection attacks.
+
+PKCE is enabled by default in FusionAuth for public clients. To enable PKCE for your confidential app, set the PKCE dropdown in your app's settings to "Required".
+
+{% include _image.liquid src="/assets/img/blogs/spring-fusionauth/fusionauth-pkce-required.png" alt="Enforcing PKCE for all client types." class="img-fluid" figure=false %}
+
+## Setting up Spring
+
+Navigate to [Spring Initialzr](https://start.spring.io), and create a new Spring Boot project with the following options set:
+
+- You can choose either Gradle or Maven as your package manager—we've used Maven for this example
+- Spring Boot 2.7.5
```

Comment:
I don't see 2.7.5 any longer there. Do we need to bump this version?

Reply:
I've changed to 2.7.x as it seems to be updated fairly frequently with minor patches